packages/x-components/src/x-modules/scroll/components/main-scroll.vue

```diff
@@ -0,0 +1,400 @@
+<script lang="ts">
+  import { Component, Prop, Watch } from 'vue-property-decorator';
+  import { xComponentMixin } from '../../../components/x-component.mixin';
+  import { State } from '../../../components/decorators/store.decorators';
+  import { XOn } from '../../../components/decorators/bus.decorators';
+  import { XProvide } from '../../../components/decorators/injection.decorators';
+  import { NoElement } from '../../../components/no-element';
+  import { scrollXModule } from '../x-module';
+  import { FirstVisibleItemObserverKey } from './scroll.const';
+  import { ScrollVisibilityObserver } from './scroll.types';
+
+  /**
+   * Extends the scroll making it able to sync the first visible element, and allowing
+   * the children position to be restored.
+   *
+   * Each child element that wants to have this support must be wrapped in a {@link MainScrollItem}
+   * component.
+   *
+   * @public
+   */
+  @Component({
+    mixins: [xComponentMixin(scrollXModule)]
+  })
+  /* eslint-disable @typescript-eslint/unbound-method */
+  export default class MainScroll extends NoElement {
+    /**
+     * The reference to the HTML node that has the scroll.
+     *
+     * @public
+     */
+    public $el!: HTMLElement;
+
+    /**
+     * If `true`, sets this scroll instance to the main of the application. Being the main
+     * scroll implies that features like restoring the scroll when the query changes, or storing
+     * the scroll position in the URL will be enabled for this container.
+     *
+     * @public
+     */
+    @Prop({ default: false, type: Boolean })
+    public useWindow!: boolean;
+
+    /**
+     * Timeout in milliseconds to abort trying to restore the scroll position to the target
+     * element.
+     *
+     * @public
+     */
+    @Prop({ default: 5000 })
+    public restoreScrollTimeoutMs!: number;
+
+    /**
+     * Intersection percentage to consider an element visible.
+     *
+     * @public
+     */
+    @Prop({ default: 0.5 })
```

Comment:
reduce the default value to 0.3 to minimize the animation issue, add a TODO if you want to remember restore it when animation issue fixed